### PR TITLE
Issue #549: Upgrade kafka-avro-serializer dependency

### LINKED ISSUE #549: Upgrade kafka-avro-serializer dependency
## Feature description 

From Zendesk ticket [#21480](https://neotechnology.zendesk.com/agent/tickets/21480) has been highlighted the need to upgrade `kafka-avro-serializer` dependency from the version 5.2.2 (which includes an old version of Netty library - 3.10.6.Final - with a security vulnerability) to the 7.2.2. 

Could this result into the needs to update also the `kafka-clients` library accordingly? Actually we're using 2.4.1, which has some vulnerabilities (see more details at the following link: https://mvnrepository.com/artifact/org.apache.kafka/kafka-clients/2.4.1)

### PR DESCRIPTION
**Pull request by mroiter-larus**
Fixes #549 

Updated versions of the following libraries:

- io.confluent.kafka.avro.serializer, from 5.2.2 to 7.4.0
- org.apache.avro.avro, from 1.8.2 to 1.11.2
- com.google.guava.guava, from 27.0.1-jre to 32.1.1-jre
- org.apache.kafka.kafka-clients and org.apache.kafka.connect-api, from 2.4.1 to 3.5.1

Fixed also a couple of compilation error in CommonExtension.kt and KafkaErrorServiceTest.kt

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -55,15 +55,15 @@
         <kotlin.version>1.6.10</kotlin.version>
         <kotlin.coroutines.version>1.6.0</kotlin.coroutines.version>
         <neo4j.version>4.4.3</neo4j.version>
-        <kafka.version>2.4.1</kafka.version>
+        <kafka.version>3.5.1</kafka.version>
```
Comment (ali-ince) [inline pom.xml:58]: Let's stick with 2.6.3 as we discussed before with @conker84.

File: pom.xml [hunk 2]
```diff
@@ -55,15 +55,15 @@
         <kotlin.version>1.6.10</kotlin.version>
         <kotlin.coroutines.version>1.6.0</kotlin.coroutines.version>
         <neo4j.version>4.4.3</neo4j.version>
```
Comment (ali-ince) [inline pom.xml:57]: let's upgrade this to the latest also?

File: pom.xml [hunk 3]
```diff
@@ -55,15 +55,15 @@
...
-        <kafka.version>2.4.1</kafka.version>
+        <kafka.version>3.5.1</kafka.version>
         <jackson.version>2.13.1</jackson.version>
```
Comment (ali-ince) [inline pom.xml:59]: this can also be upgraded to 2.15.3.

File: pom.xml [hunk 4]
```diff
@@ -55,15 +55,15 @@
...
         <kotlin.compiler.incremental>true</kotlin.compiler.incremental>
         <neo4j.java.driver.version>4.4.2</neo4j.java.driver.version>
```
Comment (ali-ince) [inline pom.xml:61]: There's 4.4.12

**Review by ali-ince — CHANGES_REQUESTED** [review]
Let's not move to the latest kafka. I also suggested to upgrade other dependencies too, but happy to see it as a separate PR if you prefer so.

File: pom.xml (hunk 4 above)
Comment (conker84) [reply pom.xml:61]: We can skip this is used by the kafka connector which is not used here anymore

File: pom.xml (hunk 1 above)
Comment (conker84) [reply pom.xml]: done

File: pom.xml (hunk 3 above)
Comment (conker84) [reply pom.xml]: done

**Comment by conker84** [comment]
@ali-ince we had to downgrade to 2.14.3 as with 2.15 was complaining about classes compiled with Java 17.

**Comment by ali-ince** [comment]
also just checked the jackson jdk compatibility and it looks like it is still on java 8? 😕

**Comment by conker84** [comment]
> also just checked the jackson jdk compatibility and it looks like it is still on java 8? 😕

Yep I checked it to, that's odd...

**Pull request merged**